### PR TITLE
Refactor identification

### PR DESCRIPTION
**Pull request by karlowich**
I have cleaned up the identification part of the documentation, such that it is more clear which idfy structs are command set specific and which aren't.

There were multiple duplicated structs these have been cleaned up.

### CODE REVIEW

File: include/libxnvme_spec.h [hunk 1]
```diff
@@ -2324,16 +2291,38 @@ struct xnvme_spec_nvm_idfy_ctrlr {
...
+			uint8_t gpistm    : 1;
+
+			/** 16b guard protection information storage tag support */
+			uint8_t gpists    : 1;
 
-	uint16_t mssrl; ///< Maximum Single Source Range Length
-	uint32_t mcl;   ///< Maximum Copy Length
-	uint8_t msrc;   ///< Maximum Source Range Count
+			/** storage tag check read support */
+			uint8_t stcrs     : 1;
```
Comment (ankit-sam) [inline include/libxnvme_spec.h:2314]: Order doesn't look right.
Bit 0 is for storage tag support, 1 is for storage tag mask and 2 is for storage tag check read support.
https://nvmexpress.org/wp-content/uploads/NVM-Express-NVM-Command-Set-Specification-1.0d-2023.12.28-Ratified.pdf

**Review by ankit-sam — COMMENTED** [review]
One comment, else looks good

File: include/libxnvme_spec.h (hunk 1 above)
Comment (karlowich) [reply include/libxnvme_spec.h:2314]: Nice catch, thanks!

**Review by ankit-sam — APPROVED** [review]
Looks good

**Pull request merged**